### ISSUE TITLE
DNS provider incorrectly replaces TXT record: never stabilizes

### ISSUE THREAD

**Issue by mike-at-spotify** [hashicorp/terraform-provider-google#9416]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

0.15.5

### Affected Resource(s)

* google_dns_record_set

### Terraform Configuration Files

`main.tf`
```tf
terraform {
  backend "gcs" {
    bucket = "alf-testing-terraform-state"
    prefix = "terraform/state/demozone-com"
  }
}

resource "google_dns_managed_zone" "google_zones" {
  dns_name = "demozone.com."
  name     = "demozone-com"
}

resource "google_dns_record_set" "this" {
  name         = "txt1.demozone.com."
  managed_zone = "demozone-com"
  type         = "TXT"
  ttl          = 300
  rrdatas      =  ["v=DKIM1; k=rsa; p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtJSh7KoLT+LnUDMK Ef1gTSwbHsMQ/VBR7v58ALwSUL173mtLvgR9/jqJmjLlJn6NcgJsPWFOO2Kh GBkH1rmTDUTl6XDRPEwXYB5SlSSTlcGRneyLq07h6nNprICMBQsOsBa2OzxQ JA92xJJZ5AGmf8xFdNihy8KwU93IQfSX69oMgvYJLV0zkxd1ErxMHoAVIBJx j/yudSowINRwpeD5JsHuWaCkRwb7+gpZiiL18VKkGMOjzdF8f2hr0ruOF8aH XpWfK/51Ob0Bd5SV5LPMSQZ4pubQUFdcKh+v4e39gLBUKZV9hEVBTFsgyrba Da+cLyf+sgGZxYu1CJ/vAo6xxQIDAQAB"]
}
```
`providers.tf`
```tf
provider "google" {
  project = "alf-testing"
}
```
`versions.tf`
```tf
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 3.73.0"
    }
  }
  required_version = ">= 0.13"
}

```
### Debug Output
https://gist.github.com/mike-at-spotify/a6e0ebc6de33eee2e2fa15798a7da3f8

### Expected Behavior

When `terraform apply` is run twice consecutively without changing the resources, the second `terraform apply` should not make any changes.

### Actual Behavior
Every `terraform apply` makes the same change in spite of the fact that the `main.tf` has not changed and also in spite of the fact that the previous `terraform apply` succeeded.

In other words, the infrastructure never becomes stable.

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`
3. `terraform apply`
4. `terraform apply`

### Important Factoids
This is about as simple as it gets.  The infrastructure should be stable after the first apply, but it isn't.

**Comment by venkykuberan** [comment]
fix for this  #9112 should take care of this issue as well

**Comment by rileykarson** [comment]
Note: the solution to https://github.com/hashicorp/terraform-provider-google/issues/9257 is almost certainly the solution to this
